service/history/workflowExecutionContext.go

```diff
@@ -46,15 +46,20 @@ const (
 
 type (
 	workflowExecutionContext interface {
-		appendHistoryEvents(history []*workflow.HistoryEvent, transactionID int64, doLastEventValidation bool) (int, error)
+		appendFirstBatchHistoryForContinueAsNew(newStateBuilder mutableState, transactionID int64) error
+		appendFirstBatchEventsForActive(msBuilder mutableState) (int, error)
+		appendFirstBatchEventsForStandby(msBuilder mutableState, history []*workflow.HistoryEvent) (int, error)
 		clear()
 		continueAsNewWorkflowExecution(context []byte, newStateBuilder mutableState, transferTasks []persistence.Task, timerTasks []persistence.Task, transactionID int64) error
+		createWorkflowExecution(
```

Comment:
why `now` is also a param?

Reply:
active side and passive side has different "now", active use the real clock, passive use the event time.